File: src/game/powerups/portal/Portal.java
package game.powerups.portal;

import game.Bomberman;
import game.entity.models.BomberEntity;
import game.entity.models.Coordinates;
import game.powerups.PowerUp;

public abstract class Portal extends PowerUp {
    public Portal(Coordinates coordinates){
        super(coordinates);
    }

    @Override
    public boolean isDisplayable() {
        return false;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        Bomberman.getMatch().toggleGameState();
    }
}

File: src/game/powerups/portal/Portal.kt
package game.powerups.portal

import game.Bomberman
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.powerups.PowerUp

abstract class Portal : PowerUp {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override val isDisplayable: Boolean = false

    override fun doApply(entity: BomberEntity) {
        Bomberman.getMatch().toggleGameState()
    }
}